src/Sylius/Component/Addressing/Model/Country.php

```diff
@@ -146,6 +146,22 @@ public function hasProvince(ProvinceInterface $province)
     /**
      * {@inheritdoc}
      */
+    public function isEqualTo($data)
+    {
+        if (is_object($data)) {
+            if (!$data instanceof CountryInterface) {
+                return false;
+            }
+
+            return $data->getIsoName() === $this->isoName || $data->getId() === $this->id;
+        }
+
+        return $data === $this->id;
```

Comment:
As said before, it's for BC, and you don't compare int to object here, please read code.